frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -44,45 +52,166 @@ export async function action({ context, request }: Route.ActionArgs) {
+          v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.trim(),
+          v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.invalid')),
+        ),
+      });
+      input.currentStatusInCanada = formData.get('currentStatusInCanada') as string;
+      const currentStatusResult = v.safeParse(currentStatusSchema, input, { lang });
 
-  if (!parsedDataResult.success) {
-    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
-  }
+      let documentTypeResult = null;
 
-  context.session.inPersonSINCase = {
-    ...(context.session.inPersonSINCase ?? {}),
-    ...input,
-  };
+      if (formData.get('step') === '2') {
+        const documentTypeSchema = v.object({
+          documentType: v.pipe(
+            v.string(t('protected:primary-identity-document.document-type.required')),
+            v.trim(),
+            v.nonEmpty(t('protected:primary-identity-document.document-type.required')),
+          ),
+        });
+        input.documentType = formData.get('documentType') as string;
+        documentTypeResult = v.safeParse(documentTypeSchema, input, { lang });
+
+        if (!documentTypeResult.success) {
+          return data(
+            {
+              errors: {
+                ...(currentStatusResult.issues ? v.flatten<typeof currentStatusSchema>(currentStatusResult.issues).nested : {}),
+                ...v.flatten<typeof documentTypeSchema>(documentTypeResult.issues).nested,
+              },
+            },
+            { status: 400 },
+          );
+        }
+      } else if (!currentStatusResult.success) {
+        return data({ errors: v.flatten<typeof currentStatusSchema>(currentStatusResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase = {
+        ...(context.session.inPersonSINCase ?? {}),
+        ...input,
+        primaryDocStep: formData.get('step') as string,
+      };
 
-  throw i18nRedirect('routes/protected/request.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+      throw i18nRedirect('routes/protected/request.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/secondary-docs.tsx"
+    }
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
 }
 
 export default function PrimaryDocs({ loaderData, actionData, params }: Route.ComponentProps) {
   const { t } = useTranslation(handle.i18nNamespace);
 
   const fetcher = useFetcher<Info['actionData']>();
   const isSubmitting = fetcher.state !== 'idle';
-  const errors = fetcher.data?.errors;
+  const errors: { currentStatusInCanada?: string[]; documentType?: string[] } | undefined = fetcher.data?.errors ?? undefined;
+
+  const [step, setStep] = useState(1);
```

Comment:
Try to avoid using magic numbers.. replace `1` with something more descriptive and meaningful.